snuba/cli/bulk_load.py

```diff
@@ -0,0 +1,36 @@
+import logging
+
+import click
+
+from snuba import settings
+from snuba.datasets.factory import get_dataset
+from snuba.snapshots.postgres_snapshot import PostgresSnapshot
+
+
+@click.command()
+@click.option('--dataset', type=click.Choice(['events', 'groupedmessage', 'outcomes']),
+              help='The dataset to bulk load')
+@click.option('--source',
+              help='Source of the dump. Depending on the dataset it may have different meaning.')
+@click.option('--dest-table',
+              help='Clickhouse destination table.')
+@click.option('--log-level', default=settings.LOG_LEVEL, help='Logging level to use.')
+def bulk_load(dataset, dest_table, source, log_level):
+    import sentry_sdk
+
+    sentry_sdk.init(dsn=settings.SENTRY_DSN)
+    logging.basicConfig(level=getattr(logging, log_level.upper()), format='%(asctime)s %(message)s')
+
+    logger = logging.getLogger('load-snapshot')
+    logger.info("Start bulk load process for dataset %s, from source %s", dataset, source)
+    dataset = get_dataset(dataset)
+
+    # TODO: Have a more abstract way to load sources if/when we support more than one.
+    snapshot_source = PostgresSnapshot.load(
+        path=source,
+        product=settings.SNAPSHOT_LOAD_PRODUCT,
+    )
```

Comment:
Sort of seems like the source should be encapsulated on the `dataset`, and not here — I'm anticipating the source type to be pretty coupled to the dataset, I'd think?

Reply:
I was quite conflicted about this. The first implementation put the responsibility on the dataset. Though the idea here is to split the source from the loader. 
The source: BulkLoadSource just provides an abstraction over a dump organized in tables. It does not even says it has to be CSV nor that it has to be in multiple files. This is not dataset specific. I would say the BulkLoader (not the BulkLoadSource) should be dataset specific together with the logic that parses the file (which is not in this PR at all).

Imagine this case:
GroupedMessage dataset. We generally fill it in from a postgres snapshot (PostgresSnapshot) through a PostgresCSVParser (which does not exists yet). But it can also be filled in from a backup or a dump we taken from clickhouse, Then we may have a ClickhouseDumpSource. 

Does this make sense ?

What is tricky and suboptimal is that this script is not able to do something like:
"I want to fill in groupedmessage dataset from a postgres snapshot"
vs
"I want to fill in groupedmessage dataset from a clickhosue file dump"

Reply:
Yeah, that does make sense. I think I jumped from "I don't think this file should literally specify `PostgresSnapshot`" (basically [echoing your comment](https://github.com/getsentry/snuba/pull/398/files/1913d8193a794c86441635dfd40103e9b014dfa9#diff-a33ad16e599ecd6955e130e48dc3cb47R28)) but you're right, that amount of coupling to the dataset wouldn't make sense either.